ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java

```diff
@@ -0,0 +1,455 @@
+        String mainMethodClassName = fileName.substring(0, fileName.length() - TEMP_FILE_SUFFIX.length());
+
+        JarResolver jarResolver = jBallerinaBackend.jarResolver();
+        ClassLoader classLoader = jarResolver.getClassLoaderWithRequiredJarFilesForExecution();
+        // First run configure initialization
+        invokeDirectMethod(classLoader, CONFIGURE_INIT_CLASS_NAME, CONFIGURE_INIT_METHOD_NAME,
+                new Class[]{Path.class}, new Object[]{CONFIG_PATH});
+        // Initialize the module
+        invokeScheduledMethod(classLoader, MODULE_INIT_CLASS_NAME, MODULE_INIT_METHOD_NAME);
+        // Then call main method
+        Object result = invokeScheduledMethod(classLoader, mainMethodClassName, MODULE_MAIN_METHOD_NAME);
+        addDiagnostic(Diagnostic.debug("Result: " + result));
+        return result == null;
+    }
+
+    /* Invocation methods */
+
+    /**
+     * Invokes a method that is in the given class.
+     * The method must be a static method accepting only one parameter, a {@link Strand}.
+     *
+     * @param classLoader Class loader to find the class.
+     * @param className   Class name with the method.
+     * @param methodName  Method name to invoke.
+     * @return The result of the invocation.
+     * @throws InvokerException If invocation failed.
+     */
+    protected Object invokeScheduledMethod(ClassLoader classLoader, String className, String methodName)
+            throws InvokerException {
+        try {
+            addDiagnostic(Diagnostic.debug(String.format("Running %s.%s on schedule", className, methodName)));
+
+            // Get class and method references
+            Class<?> clazz = classLoader.loadClass(className);
+            Method method = clazz.getDeclaredMethod(methodName, Strand.class);
+            Function<Object[], Object> methodInvocation = createInvokerCallback(method);
+
+            // Schedule and run the function and return if result is valid
+            BFuture out = scheduler.schedule(new Object[1], methodInvocation, null, null, null,
+                    PredefinedTypes.TYPE_ERROR, null, null);
+            scheduler.start();
+
+            Object result = out.getResult();
+            Throwable panic = out.getPanic();
+            addDiagnostic(Diagnostic.debug("Panic: " + panic));
+            addDiagnostic(Diagnostic.debug("Result: " + result));
+
+            if (panic != null) {
+                throw new InvokerException(panic);
+            }
+            return result;
+        } catch (ClassNotFoundException e) {
+            addErrorDiagnostic(className + " class not found: " + e.getMessage());
+            throw new InvokerException(e);
+        } catch (NoSuchMethodException e) {
+            addErrorDiagnostic(methodName + " method not found: " + e.getMessage());
+            throw new InvokerException(e);
+        } catch (RuntimeException e) {
+            addErrorDiagnostic("Unexpected error: " + e.getMessage());
```

Comment:
We are not supposed to catch `RuntimeException` but they should be fixed as much as possible. Where can we expect `RuntimeException` in this code? Can we fix them without catching them?

Reply:
Shall we fix in all places if it is applicable?

Reply:
The runtime exception can occur and from the user snippet as well. Here, and in the next position, we try to invoke a method using reflection and it can throw a RuntimeException. IMO if that happens, a REPL should continue to work without crashing and just notify the user that something went wrong.

Reply:
I think if  `RuntimeException` is thrown inside a method call it is propagated as `InvocationTargetException`. I think we need to catch that instead of the `RuntimeException`.

Reply:
I'll remove the handlers for `RuntimeException` then :)